scripts/PyChop/ISISDisk.py

```diff
@@ -0,0 +1,384 @@
+                    elif len(frequency) == 5:
+                        self.freq = frequency
+                    else:
+                        raise ValueError('Frequency must be a 1-, 2- or 5-element list/array')
+                else:
+                    self.freq = [frequency/4., 10., frequency/2., frequency/2., frequency]
+            if 'Chopper2Phase' in kwargs.keys():
+                self.Chop2Phase = kwargs['Chopper2Phase']
+        elif 'MERLIN' in self.instname:
+            if hasattr(frequency, "__len__"):
+                self.freq = [50., frequency[0]]
+            else:
+                self.freq = [50., frequency]
+            if 'Chopper2Phase' in kwargs.keys():
+                self.Chop2Phase = kwargs['Chopper2Phase']
+        else:
+            raise RuntimeError('Instrument name has not been set')
+
+    def getFrequency(self):
+        """
+        Returns a list of the chopper frequencies in Hz.
+        """
+        return self.freq
+
+    def setEi(self, Ei):
+        """
+        Sets the focused incident energy in meV
+        """
+        if Ei > 0:
+            self.Ei = Ei
+        else:
+            raise ValueError('Incident energy must be greater than zero')
+
+    def getEi(self):
+        """
+        Returns the focussed incident energy in meV
+        """
+        return self.Ei
+
+    def __LETgetResolution(self, single_mode, Etrans=None, Ei_in=None):
+        """
+        Private method to calculate the resolution at given Ei, Etrans from chopper opening times.
+        """
+        Ei = self.Ei if Ei_in is None else Ei_in
+        instpars = [self.dist, self.nslot, self.slot_width, self.guide_width, self.radius, self.numDisk,
+                    self.samp_det, self.chop_samp, self.source_rep, self.tmod, self.frac_ei, self.ph_ind]
+        Eis, _, chop_times, lastChopDist, _ = MulpyRep.calcChopTimes(Ei, self.freq, instpars, self.Chop2Phase)
+        res_el, percent, chop_width, mod_width = MulpyRep.calcRes(Eis, chop_times, lastChopDist, self.chop_samp, self.samp_det)
+        if single_mode:
+            #ie_list = [ii for ii,ee in enumerate(Eis) if np.abs((ee-Ei)/Ei)<0.05]
+            ie_list = np.where(np.abs(np.array(Eis)-Ei) == np.min(np.abs(np.array(Eis)-Ei)))[0]
+            Et = np.linspace(0.05, 0.95*Ei, 19, endpoint=True) if (Etrans is None) else Etrans
+        else:
+            ie_list = range(len(Eis))
+            # This is the relative energy transfer
+            Et = np.linspace(0.05, 0.95, 19, endpoint=True) if (Etrans is None) else Etrans
+        res_list = []
+        if not np.shape(Et):
+            Et = [Et]
+        for ie in ie_list:
+            t_mod_chop = 252.82 * lastChopDist * np.sqrt(81.81/Eis[ie])
+            res = []
+            for en in Et:
+                Ef = Eis[ie] - en
+                fac = (Ef/Eis[ie])**1.5
+                chopRes = (2*chop_width[ie]/t_mod_chop) * ((self.samp_det+self.chop_samp+lastChopDist)/self.samp_det)*fac
```

Comment:
There is a missing term in this formula. It should read 
`chopRes = (2*chop_width[ie]/t_mod_chop) *(1+ ((self.samp_det+self.chop_samp+lastChopDist)/self.samp_det)*fac)` Same error in the documentation. I will attach the derivation